### PR TITLE
Deep fried inedible items are moved to stomach when eaten

### PR DESCRIPTION
**Pull request by cacogen**
## About The Pull Request

AnturK said it was okay to make another PR for https://github.com/tgstation/tgstation/pull/64747.

### Quick rundown/basic gestalt:
- Deep fried inedible items are moved to stomach when eaten, instead of being deleted
- Having inedible items in stomach increases disgust and damages stomach at random intervals based on their total size
- Vomiting empties stomach
- Stomachs act like (worthless) bags now
- Default stomachs hold a combined weight class of 12 (only for consumed items), each tier of cybernetic stomach carries more
- Toy singulo suicide has been updated to use the stomach's contents instead of cavity implants

### Some considerations:

**Q: Can this be used to create infinite food?**
A: No, vomiting causes more nutrition loss than you can offset with a few deep fried inedible objects.

**Q: Can this be used to carry the armoury?**
A: Vomiting stuns you, so the other person would get access to the vomited guns first. You could use a monkey with the tier 3 stomach to carry 6 bulky guns, but they'd vomit them up in less than a minute.

**Q: Given organs are small items (fit in boxes), does this make stomachs better storage items than smuggler's satchels?**
A: No, stomachs only hold two tiny items. Eating ignores this limit.

**Q: Does this mean I could forcefeed a monkey deep fried shotguns, remove its stomach, and carry it around in my box?**
A: No, removing stomachs drop their contents.

**Q: Does this affect stomachs of species that don't eat?**
A: Species that don't eat can still eat, so I've only removed it from Ethereal stomachs. They're called biological batteries and given they don't eat I find it hard to justify something called that holding eaten things. The rest are essentially still stomachs.

**Q: Aren't deep fried objects unusable?**
A: The deep fry holder, which is an object that mimics the deep fried object, contains the deep fried object. Before this PR, the deep fried object was deleted when the deep fry holder was eaten. Now, the object is moved to the stomach.

**Q: What happens when mobs are consumed?**
A: This works similarly to being held as a mob. You see and hear what the mob holding you sees. You can be heard. Eventually you're vomited back up. The alternative is preventing objects with clients in them from being deep fried, but that's boring.

## Why It's Good For The Game

It's cool and adds more depth  

## Changelog

:cl:
balance: Deep fried inedible items are moved to stomach when eaten, instead of being deleted
add: Having inedibles in stomach increases disgust and damages stomach at random intervals based on their total volume
add: Having too many items in stomach triggers vomiting
add: Vomiting empties stomach
add: Stomachs act like (worthless) bags
expansion: Toy singulo suicide uses stomach contents instead of cavity implants
/:cl:

### CODE REVIEW

**Comment by Y0SH1M4S73R** [comment]
You know what would be a hilarious follow-up to this PR? A pica mutation that prevents the disgust gain and/or the stomach damage.

**Comment by nianjiilical** [comment]
make sure stomach contents are checked for steal objectives so we can just eat the hand teleporter

**Comment by Ryll-Ryll** [comment]
Wow, this is a far, far more fleshed out PR than it was I first looked at the old one. Very interested!

**Comment by ErdinyoBarboza** [comment]
full circle back to vorecode

File: code/modules/surgery/organs/stomach/_stomach.dm [hunk 1]
```diff
@@ -39,6 +43,31 @@
 	else
 		reagents.flags |= REAGENT_HOLDER_ALIVE
 
+/obj/item/organ/stomach/ComponentInitialize()
```
[the remark this answers is not in the document]
Comment (cacogen) [reply code/modules/surgery/organs/stomach/_stomach.dm:46]: So, what should I use instead?

**Comment by TemporalOroboros** [comment]
Along the same lines as the singularity toy suicide you might want to take a look at the grenade suicide.

File: code/modules/surgery/organs/stomach/_stomach.dm (hunk 1 above)
[the remark this answers is not in the document]
Comment (MrMelbert) [reply code/modules/surgery/organs/stomach/_stomach.dm:46]: You can just put it in `Initialize()`

**Comment by oklews** [comment]
aren't deepfried objects unusable? Are you like removing the deepfried status from it when it exits the stomach? Could you make eating sharp objects cause bleeding?

**Comment by oklews** [comment]
> aren't deepfried objects unusable? Are you like removing the deepfried status from it when it exits the stomach? Could you make eating sharp objects cause bleeding?

Local man doesn't read the PR

**Comment by oklews** [comment]
would you be open to making formerly undeepfriable things deepfriable?

**Comment by cacogen** [comment]
I dunno, making things like the nuke disk consumable seems like it's for a different PR. There are still issues with this one that mean it might not be merged, namely the consequences of putting the stomach in the mob or how limited the feature becomes if I have to leave the organ in nullspace. You couldn't eat the disk anyway if I had to do that, because it would immediately move.

**Comment by interestingusernam3** [comment]
vore

**Comment by oklews** [comment]
> I dunno, making things like the nuke disk consumable seems like it's for a different PR. There are still issues with this one that mean it might not be merged, namely the consequences of putting the stomach in the mob or how limited the feature becomes if I have to leave the organ in nullspace. You couldn't eat the disk anyway if I had to do that, because it would immediately move.

Things like the HOS laser or the Captain's laser aren't deep friable

**Comment by cacogen** [comment]
I'm going to close this because I have too many PRs to focus on at the moment and I don't think this is going to be merged because of the potential for bugs from putting the stomach into the mob.

**Pull request closed without merging**